packages/syft/src/syft/client/domain_client.py

```diff
@@ -123,25 +123,35 @@ def upload_dataset(self, dataset: CreateDataset) -> SyftSuccess | SyftError:
             )
             prompt_warning_message(message=message, confirm=True)
 
-        for asset in tqdm(dataset.asset_list, colour="green"):
-            print(f"Uploading: {asset.name}")
-            try:
-                twin = TwinObject(
-                    private_obj=asset.data,
-                    mock_obj=asset.mock,
-                    syft_node_location=self.id,
-                    syft_client_verify_key=self.verify_key,
-                )
-                twin._save_to_blob_storage()
-            except Exception as e:
-                return SyftError(message=f"Failed to create twin. {e}")
-            response = self.api.services.action.set(twin)
-            if isinstance(response, SyftError):
-                print(f"Failed to upload asset\n: {asset}")
-                return response
-            asset.action_id = twin.id
-            asset.node_uid = self.id
-            dataset_size += get_mb_size(asset.data)
+        with tqdm(
+            total=len(dataset.asset_list), colour="green", desc="Uploading"
+        ) as pbar:
+            for asset in dataset.asset_list:
+                try:
+                    twin = TwinObject(
+                        private_obj=asset.data,
+                        mock_obj=asset.mock,
+                        syft_node_location=self.id,
+                        syft_client_verify_key=self.verify_key,
+                    )
+                    twin._save_to_blob_storage()
```

Comment:
We want to only call `_save_to_blob_storage()` when the object's size is more than 16 Mb. 
Link to the issue: https://github.com/OpenMined/Heartbeat/issues/1274